pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -308,6 +308,7 @@ func (o *SyncOptions) applySyncTarget(ctx context.Context, kcpClient kcpclient.I
 		if err != nil && !apierrors.IsAlreadyExists(err) {
 			return nil, fmt.Errorf("failed to create synctarget %q: %w", syncTargetName, err)
 		}
+		return syncTarget, nil
```

Comment:
What is this change here expected to fix ?

Reply:
it should return early after create, otherwise will go to the the unnecessary following patch logic, and it may fail because of version conflict.

Reply:
But it also returns in case the creation failed because of an `AlreadyExists` error, isn't it ? Should it ?